lib/spack/spack/build_systems/python.py

```diff
@@ -8,93 +8,22 @@
 import llnl.util.tty as tty
-from llnl.util.filesystem import (
-    filter_file,
-    find,
-    find_all_headers,
-    find_libraries,
-    is_nonsymlink_exe_with_shebang,
-    path_contains_subdirectory,
-    same_path,
-    working_dir,
-)
-from llnl.util.lang import classproperty, match_predicate
-
-from spack.directives import depends_on, extends
+
+import spack.builder
+import spack.multimethod
+import spack.package_base
+from spack.directives import buildsystem, depends_on, extends
 from spack.error import NoHeadersError, NoLibrariesError, SpecError
-from spack.package_base import PackageBase, run_after
 from spack.version import Version
 
+from ._checks import BaseBuilder
 
-class PythonPackage(PackageBase):
-    """Specialized class for packages that are built using pip."""
-
-    #: Package name, version, and extension on PyPI
-    pypi = None  # type: Optional[str]
-
-    maintainers = ["adamjstewart", "pradyunsg"]
-
-    # Default phases
-    phases = ["install"]
-
-    # To be used in UI queries that require to know which
-    # build-system class we are using
-    build_system_class = "PythonPackage"
-
-    #: Callback names for install-time test
-    install_time_test_callbacks = ["test"]
-
-    extends("python")
-    depends_on("py-pip", type="build")
-    # FIXME: technically wheel is only needed when building from source, not when
-    # installing a downloaded wheel, but I don't want to add wheel as a dep to every
-    # package manually
-    depends_on("py-wheel", type="build")
 
-    py_namespace = None  # type: Optional[str]
-
-    @staticmethod
-    def _std_args(cls):
-        return [
-            # Verbose
-            "-vvv",
-            # Disable prompting for input
-            "--no-input",
-            # Disable the cache
-            "--no-cache-dir",
-            # Don't check to see if pip is up-to-date
-            "--disable-pip-version-check",
-            # Install packages
-            "install",
-            # Don't install package dependencies
-            "--no-deps",
-            # Overwrite existing packages
-            "--ignore-installed",
-            # Use env vars like PYTHONPATH
-            "--no-build-isolation",
-            # Don't warn that prefix.bin is not in PATH
-            "--no-warn-script-location",
-            # Ignore the PyPI package index
-            "--no-index",
-        ]
-
-    @classproperty
-    def homepage(cls):
-        if cls.pypi:
-            name = cls.pypi.split("/")[0]
-            return "https://pypi.org/project/" + name + "/"
-
-    @classproperty
-    def url(cls):
-        if cls.pypi:
-            return "https://files.pythonhosted.org/packages/source/" + cls.pypi[0] + "/" + cls.pypi
-
-    @classproperty
-    def list_url(cls):
-        if cls.pypi:
-            name = cls.pypi.split("/")[0]
-            return "https://pypi.org/simple/" + name + "/"
+class PythonExtension(spack.package_base.PackageBase):
+    maintainers = ["adamjstewart"]
```

Comment:
Completely off topic, but it would be cool if `maintainers` was a directive that added to a list instead of a variable that could be overwritten in subclasses.

Reply:
We probably need both modes to deal with inheriting from another package (e.g. `amd-fftw` from `fftw`)